src/status_im2/contexts/chat/bottom_sheet_composer/actions/view.cljs

```diff
@@ -64,13 +64,57 @@
             :i/arrow-up]])]))])
 
 (defn audio-button
-  []
-  [quo/button
-   {:on-press #(js/alert "to be added")
-    :icon     true
-    :type     :outline
-    :size     32}
-   :i/audio])
+  [{:keys [record-permission? audio-file]}
+   {:keys [recording? gesture-enabled?]}
+   {:keys [container-opacity]}]
+  [rn/view
+   {:style          (style/record-audio-container)
+    :pointer-events :box-none}
+   [quo/record-audio
+    {:record-audio-permission-granted    record-permission?
+     :on-start-recording                 (fn []
+                                           (reset! recording? true)
```

Comment:
@briansztamfater I am not sure what was `on-init` function doing? I tried removing it and did not notice any issue. Please let me know if it is really needed.

Reply:
It is used to save the reset function in `record-audio-reset-fn` atom from the audio component and reset the state on show-send / hide-send functions in current composer implementation. One of the reasons, for example, let's say a user is replying to a message and recording an audio while the keyboard is being shown. Then, in the middle of the recording, the user starts typing a message, causing the the recorder to be hidden, and `@record-audio-reset-fn` to be called to reset current audio component to its initial state.

Reply:
Got it! Thank you @briansztamfater